src/translation/Translation.ts

```diff
@@ -0,0 +1,45 @@
+import trJson from './tr.json'
+
+export class Translation {
+    data: { [index: string]: string }
+    fallback: { [index: string]: string }
+
+    constructor(data: { [index: string]: string }, fallback: { [index: string]: string }) {
+        if (!data) throw Error('Translation is undefined')
+        if (!fallback) throw Error('Translation fallback is undefined')
+        this.data = data
+        this.fallback = fallback
+    }
+
+    get(key: string, parameters?: string[]): string {
+        var str: string = this.data[key]
+        if (!str) {
+            str = this.fallback[key]
+            if (!str) return key
+        }
+        if (parameters)
+            for (let i = 0; i < parameters.length; i++) {
+                str = str.replace('%' + (i + 1) + '$s', parameters[i])
+            }
+        return str || ''
+    }
+}
+
+let translation: Translation
+export function setTranslation(lang: string, overwrite = false): Translation {
+    if (translation && !overwrite) throw new Error('translation already initialized')
+
+    let json = trJson as Record<string, any>
+    for (let property in json) {
+        if (property.startsWith(lang)) {
+            translation = new Translation(json[property], json['en_US'])
```

Comment:
Also here `'en-US'`?

Reply:
Hm, but many other languages also use underscore?!

Reply:
I think we need some matching like we have somewhere in old gh maps: https://github.com/graphhopper/graphhopper-maps/issues/97